tools/triage/filters.py

```diff
@@ -0,0 +1,178 @@
+
+    @staticmethod
+    def from_json(jira_client, json):
+        """Examples:
+        {
+          "labels":{"mylabel":{"MGMT-0000":"My message"}},
+          "signatures":{"my_signature_name":{"MGMT-00001":"My signature message"}}
+        }
+        """
+        filter_params = []
+        for filter_type, items in json.items():
+            for identifier, root_issues in items.items():
+                for root_issue_key, message in root_issues.items():
+                    filter_params.append((filter_type, identifier, root_issue_key, message))
+        return Filters._get_filters(jira_client, filter_params)
+
+    @staticmethod
+    def from_args(jira_client, args):
+        """Examples:
+        <type>:<identifier>:<root issue key>:<message>
+        signatures:master_failed_to_pull_ignition_signature:MGMT-11089:master nodes failed to pull ignition
+        labels:FEATURE_myfeature:MGMT-12345:auto close issues with myfeature
+        """
+        filter_params = []
+        for filter_argument in args.filter:
+            assert filter_argument.count(":") in (2, 3)
+            filters = filter_argument.split(":")
+            # backward compatibility: assume type is signature if 3 elements only
+            if filter_argument.count(":") == 2:
+                filters.insert(0, "signature")
+            filter_params.append(filters)
+
+        return Filters._get_filters(jira_client, filter_params)
+
+    @staticmethod
+    def _get_filters(jira_client, filter_params):
+        filters = []
+        for filter_type, identifier, root_issue_key, message in filter_params:
+            filters.append(Filters.create_filter(jira_client, filter_type, identifier, root_issue_key, message))
+        return filters
+
+    @staticmethod
+    def get_triage_issues(issues, filters):
+        for issue in issues:
+            for triage_filter in filters:
+                if triage_filter.matches(issue):
+                    yield TriageIssue(issue=issue, root_issue=triage_filter.root_issue, message=triage_filter.message)
+                    continue
+
+    @staticmethod
+    def close_triage_issues(jira_api, triage_issues):
+        for issue in triage_issues:
+            jira_api.link_and_close_issue(issue.issue, issue.root_issue)
+
+    @staticmethod
+    def close_issues_by_filters(jira_api, issues, filters):
+        triage_issues = Filters.get_triage_issues(issues, filters)
+        Filters.close_triage_issues(jira_api, triage_issues)
+
+    @staticmethod
+    def close_issues_from_json(jira_client, issues, json):
+        filters = Filters.from_json(jira_client, json)
+        jira_api = JiraAPI(jira_client, logger)
+        Filters.close_issues_by_filters(jira_api, issues, filters)
+
+    @staticmethod
+    def close_issues_from_args(jira_client, issues, args):
+        filters = Filters.from_args(jira_client, args)
+        jira_api = JiraAPI(jira_client, logger)
+        Filters.close_issues_by_filters(jira_api, issues, filters)
```

Comment:
I prefer the top level function to create the relevant filters (either from args or from json or from any other source) and provide them to a function of ``Filters.close_issues_by_filters`` to actually do the closing